main.py

```diff
@@ -115,6 +115,17 @@ def handle_payload(jsonData, is_retry_payload=False):
                 and not is_retry_payload
             ):
                 handle_user_custom_field_data(jsonData)
+
+            # Handle groups and fields
+            if (
+                "fields"
+                and "groups" in jsonData["contact"]
+                and jsonData["contact"]["fields"]
+                and jsonData["contact"]["groups"] is not None
```

Comment:
Why this should not run when we got fields but not groups or vice versa? In thing it should work if any of the one this is present, no?

Reply:
Yes, got the point if the field or group is present it should be working.